id-repository/id-repository-core/src/main/java/io/mosip/idrepository/core/manager/CredentialServiceManager.java

```diff
@@ -617,25 +626,38 @@ private List<HandleInfoDTO> getHandles(String uin, IntFunction<String> saltRetre
+		return handleRepo.findByUinHash(uinHash);
+	}
 
+	private List<HandleInfoDTO> getHandlesInfo(List<Handle> handles, List<String> partnerIds,
+			IntFunction<String> saltRetreivalFunction, Consumer<EventModel> idaEventModelConsumer) {
 		List<HandleInfoDTO> handleInfoDTOS = new ArrayList<>();
-		for(Handle entity : list) {
-			HandleInfoDTO handleInfoDTO = new HandleInfoDTO();
-			String encryptSalt = uinEncryptSaltRepo
-					.retrieveSaltById(Integer.valueOf(io.mosip.kernel.core.util.StringUtils.substringBefore(entity.getHandle(), SPLITTER)));
-			try {
-				handleInfoDTO.setHandle(new String(securityManager.decryptWithSalt(
-						CryptoUtil.decodeURLSafeBase64(io.mosip.kernel.core.util.StringUtils.substringAfter(entity.getHandle(), SPLITTER)),
-						CryptoUtil.decodePlainBase64(encryptSalt), uinRefId)));
-
-				handleInfoDTO.setAdditionalData(securityManager.getIdHashAndAttributesWithSaltModuloByPlainIdHash(handleInfoDTO.getHandle(),
-						saltRetreivalFunction));
-				handleInfoDTO.getAdditionalData().put("idType", IdType.HANDLE.getIdType());
-				handleInfoDTOS.add(handleInfoDTO);
-			} catch (IdRepoAppException e) {
-				mosipLogger.error(IdRepoSecurityManager.getUser(), SEND_REQUEST_TO_CRED_SERVICE, "getHandles",
-						"\n Failed to decrypt handle due to " + e.getMessage());
+		if (handles != null && !handles.isEmpty()) {
+			for(Handle entity : handles) {
+				if (HandleStatusLifecycle.ACTIVATED.name().equals(entity.getStatus())) {
+					HandleInfoDTO handleInfoDTO = new HandleInfoDTO();
+					String encryptSalt = uinEncryptSaltRepo
+							.retrieveSaltById(Integer.valueOf(io.mosip.kernel.core.util.StringUtils.substringBefore(entity.getHandle(), SPLITTER)));
+					try {
+						handleInfoDTO.setHandle(new String(securityManager.decryptWithSalt(
+								CryptoUtil.decodeURLSafeBase64(io.mosip.kernel.core.util.StringUtils.substringAfter(entity.getHandle(), SPLITTER)),
+								CryptoUtil.decodePlainBase64(encryptSalt), uinRefId)));
+	
+						handleInfoDTO.setAdditionalData(securityManager.getIdHashAndAttributesWithSaltModuloByPlainIdHash(handleInfoDTO.getHandle(),
+								saltRetreivalFunction));
+						handleInfoDTO.getAdditionalData().put("idType", IdType.HANDLE.getIdType());
+						handleInfoDTOS.add(handleInfoDTO);
+					} catch (IdRepoAppException e) {
+						mosipLogger.error(IdRepoSecurityManager.getUser(), SEND_REQUEST_TO_CRED_SERVICE,
+								"getHandlesInfo", "\n Failed to decrypt handle due to " + e.getMessage());
+					}
+				} else if (HandleStatusLifecycle.DELETE.name().equals(entity.getStatus())) {
+					List<EventModel> eventList = new ArrayList<>();
+					eventList.addAll(createIdaEventModel(IDAEventType.REMOVE_ID, null, null, partnerIds, null,
+							entity.getHandleHash()).collect(Collectors.toList()));
+					sendEventsToIDA(eventList, IDAEventType.REMOVE_ID, idaEventModelConsumer);
+					handleRepo.updateStatus(entity.getHandleHash(), HandleStatusLifecycle.DELETE_REQUESTED.name());
```

Comment:
instead of marking the status inside the loop, can we update the status outside the loop using uinhash and current status.

Reply:
Fixes done in https://github.com/mosip/id-repository/pull/964 PR.